meson.build

```diff
@@ -34,6 +34,32 @@ if host_machine.system() == 'windows'
   src += win.compile_resources('win32/zlib1.rc', args : win_args)
 endif
 
+if get_option('asm')
+  if host_machine.cpu_family().startswith('x86')
+    if cc.get_id() == 'msvc'
+      if host_machine.cpu_family() == 'x86'
+        src += files(
+          'contrib/masmx86/inffas32.asm',
+          'contrib/masmx86/match686.asm',
+        )
+      else
+        src += files(
+          'contrib/masmx86/inffasx64.asm',
+          'contrib/masmx86/gvmat64.asm',
+        )
+      endif
```

Comment:
Did you test that this works? IIRC `cl.exe` does not support compilation of `.asm` files. You have to call `ml.exe` for that and then link it in with `link.exe`